src/components/widgets/ConstructionSiteWidget.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useContext, useEffect } from 'react';
+import { useQuery } from 'react-apollo';
+import { StyleSheet, View } from 'react-native';
+import { NavigationScreenProp } from 'react-navigation';
+
+import { colors, normalize } from '../../config';
+import { ConstructionSiteContext } from '../../ConstructionSiteProvider';
+import { filterForValidConstructionSites, graphqlFetchPolicy } from '../../helpers';
+import { constructionSite } from '../../icons';
+import { NetworkContext } from '../../NetworkProvider';
+import { getQuery, QUERY_TYPES } from '../../queries';
+import { Icon } from '../Icon';
+import { BoldText, RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Wrapper } from '../Wrapper';
+
+type Props = {
+  navigation: NavigationScreenProp<never>;
+};
+
+export const ConstructionSiteWidget = ({ navigation }: Props) => {
+  const { constructionSites, setConstructionSites } = useContext(ConstructionSiteContext);
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime: undefined });
+
+  const { data } = useQuery(getQuery(QUERY_TYPES.PUBLIC_JSON_FILE), {
+    variables: { name: 'constructionSites' },
+    fetchPolicy,
+    skip: !isConnected || !isMainserverUp
```

Comment:
why do you added `skip` here? if the fetch policy is `cache-only` based on `isConnected` and `isMainserverUp` the query should work as expected, or am i wrong?
do we need to add that `skip` in every other query now?

Reply:
you are right. the skip is unnecessary here, and would actually cause the cached version in the case of the user having no connection to not be used either.

i removed the skip in d4af0ca08d76b2fca85d3b9d39a6f148eeabf4c9